### PR TITLE
Add `python-cityhash` to CI environments

### PR DESCRIPTION
**Pull request by charlesbluca**
Now that `python-cityhash` is on conda-forge, we should be good to add this to our CI environments along with the other hasher libraries.

- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by charlesbluca** [comment]
The 3.11 packages just got published, so looks like builds are failing there - will bump this later once conda-forge's CDNs have updated

**Review by jrbourbeau — COMMENTED** [review]
Nice LGTM once CI is passing

**Comment by charlesbluca** [comment]
@jrbourbeau 3.11 packages are getting picked up and CI is passing now

**Review by jrbourbeau — APPROVED** [review]
Thanks @charlesbluca

**Comment by phofl** [comment]
@charlesbluca It looks like there aren't any osx arm packages. Either wrong name or missing. Environment can't be solved right now on my Mac. Can you take a look at adding the packages?

**Comment by charlesbluca** [comment]
Added `python-cityhash` to the OSX arch migrator in https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4365; once that's merged in we can see if there are any blockers to building the ARM packages.

Should I remove `python-cityhash` from the environments until we're sure we can build/publish for OSX ARM?

**Comment by phofl** [comment]
Do you have any idea how long it will take till the arm packages are published?

**Comment by charlesbluca** [comment]
Ended up just doing this manually in https://github.com/conda-forge/python-cityhash-feedstock/pull/6, just merged in so the packages should be published in the next hour or so

**Comment by charlesbluca** [comment]
Looks like the packages have been up for some time now, @phofl do you (or someone else with access to an OSX ARM machine) mind pulling in the packages and verifying that [dask/tests/test_hashing.py](https://github.com/dask/dask/blob/main/dask/tests/test_hashing.py) passes locally?

**Comment by phofl** [comment]
Thanks for addressing so quickly. I can solve the environment locally, so should work again

**Pull request merged**
